node/pkg/driver/node_test.go

```diff
@@ -141,224 +128,363 @@ func TestNodeStageVolume(t *testing.T) {
 	}
 }
 
-func newTestNodeService(nodeUtils driver.NodeUtilsInterface) driver.NodeService {
-	return driver.NodeService{
-		Hostname:   "test-host",
-		ConfigYaml: driver.ConfigFile{},
-		NodeUtils:  nodeUtils,
+func TestNodePublishVolume(t *testing.T) {
+	fsTypeXfs := "ext4"
+	targetPath := "/test/path"
+	targetPathWithHostPrefix := driver.GetPodPath(targetPath)
+	targetPathParentDirWithHostPrefix := filepath.Dir(targetPathWithHostPrefix)
+	stagingTargetPath := path.Join("/test/staging", driver.StageInfoFilename)
+	stagingTargetFile := path.Join(stagingTargetPath, ".stageInfo.json")
+	deviceName := "fakedev"
+	stagingInfo := map[string]string{"mpathDevice": deviceName}
+	mpathDevice := filepath.Join(device_connectivity.DevPath, deviceName)
+	targetMountPoint := &mount.MountPoint {
+		Path: targetPathWithHostPrefix,
+	}
+	fakeMountPoint := &mount.MountPoint {
+		Path: "/fake_mountpoint",
+	}
+	// mount point for which node will say that mount already exists
+	positiveMountPoint := []mount.MountPoint{*targetMountPoint, *fakeMountPoint}
+	// mount point for which node will say that mount does not exists
+	negativeMountPoint := []mount.MountPoint{*fakeMountPoint}
+	fsVolCap := &csi.VolumeCapability{
+		AccessType: &csi.VolumeCapability_Mount{
+			Mount: &csi.VolumeCapability_MountVolume{FsType: fsTypeXfs},
+		},
+		AccessMode: &csi.VolumeCapability_AccessMode{
+			Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
+		},
+	}
+	rawBlockVolumeCap := &csi.VolumeCapability{
+		AccessMode: &csi.VolumeCapability_AccessMode{
+			Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
+		},
+		AccessType: &csi.VolumeCapability_Block{
```

Comment:
why are `AccessType` and `AccessMode` declared in the opposite order, in comparison with the `fsVolCap`?